src/dash-playlist-loader.js

```diff
@@ -295,6 +243,111 @@ export default class DashPlaylistLoader extends EventTarget {
     }
   }
 
+  requestErrored_(err, request, startingState) {
+    // disposed
+    if (!this.request) {
+      return true;
+    }
+
+    // pending request is cleared
+    this.request = null;
+
+    if (err) {
+      // use the provided error or create one
+      // see requestSidx_ for the container request
+      // that can cause this.
+      this.error = typeof err === 'object' && !(err instanceof Error) ? err : {
+        status: request.status,
+        message: 'DASH request error at URL: ' + request.uri,
+        response: request.response,
+        // MEDIA_ERR_NETWORK
+        code: 2
+      };
+      if (startingState) {
+        this.state = startingState;
+      }
+
+      this.trigger('error');
+      return true;
+    }
+  }
+
+  /**
+   * Verify that the container of the sidx segment can be parsed
+   * and if it can, get and parse that segment.
+   */
+  addSidxSegments_(playlist, startingState, cb) {
```

Comment:
All of the logic from `sidxRequestFinished` and `requestSidx` merged into one because they were really doing the same things anyway. I also added some extra logic here for things that we were doing in 3+ places outside of this function, but might as well have shared.